test/user.test.js

```diff
@@ -1763,6 +1763,367 @@ describe('User', function() {
+        });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `replaceAttributes`', function(done) {
+        async.series([
+          function(next) {
+            User.findById(usersId, function(err, userFound)  {
+              if (err) return next(err);
+              userFound.replaceAttributes({ 'email': currentEmailCredentials.email,
+            'password': currentEmailCredentials.password,
+             'firstName': 'Candy' }, function(err, userInstance) {
+                if (err) return next(err);
+                assert.equal(userInstance.firstName, 'Candy');
+                next();
+              });
+            });
+          },
+          function(next) {
+            AccessToken.find({ where: { userId: usersId }}, function(err, tokens) {
+              if (err) return next(err);
+              expect(tokens.length).to.equal(2);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `updateOrCreate`', function(done) {
+        async.series([
+          function(next) {
+            User.updateOrCreate({ id: usersId, 'firstName': 'Amir',
+          'email': currentEmailCredentials.email, 'password': 'bar' },
+          function(err, userInstance) {
+            if (err) return next(err);
+            assert.equal(userInstance.firstName, 'Amir');
+            next();
+          });
+          },
+          function(next) {
+            AccessToken.find({ where: { userId: usersId }}, function(err, tokens) {
+              if (err) return next(err);
+              expect(tokens.length).to.equal(2);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `replaceById`', function(done) {
+        async.series([
+          function(next) {
+            User.replaceById(usersId, { 'firstName': 'Miroslav',
+          'email': currentEmailCredentials.email, 'password': currentEmailCredentials.password },
+            function(err, userInstance) {
+              if (err) return next(err);
+              assert.equal(userInstance.email, currentEmailCredentials.email);
+              next();
+            });
+          },
+          function(next) {
+            AccessToken.find({ where: { userId: usersId }}, function(err, tokens) {
+              if (err) return next(err);
+              expect(tokens.length).to.equal(2);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+    });
+
+    describe('User not chaning email property', function() {
+      var user1Id, user2Id, user3Id;
+      it('keeps sessions untouched if email property is not changed', function(done) {
```

Comment:
To me, this looks the same as "keeps sessions AS IS if firstName is added using {method name}".  What's the difference?

From looking at the code, it looks like you are checking that the we invalidate only sessions of the current user, right? The test name should reflect that.